specs/upgrades.md

```diff
@@ -0,0 +1,387 @@
+# IPC Spec - Upgrades
+
+As IPC evolves, upgrades are necessary to introduce new features, enhance security, improve scalability, and address new challenges. However, it is critical that all upgrades seamlessly transition without causing forks or consensus failures. This ensures that all nodes in the network remain synchronized.
+
+In this document, we describe how IPC achieves full upgradability of all its components and ensures seamless progression of the blockchain network without compromising consensus.
+
+# IPC Actor upgrades
+
+The on-chain logic for IPC is implemented in two contracts, the IPC gateway and the subnet actor, as well as an auxiliary contract used for UX purposes, the registry.
+
+The gateway is a singleton actor that lives in every IPC subnet and implements the common logic for IPC. It is responsible for managing the collateral of subnet, enforcing the firewall requirement, and the cross-net interactions.
+
+The subnet actor is a user-defined actor that implements the specific logic of a subnet. This contract is deployed in the parent from which the child subnet wants to be deployed. There is one subnet-actor for each child subnet in the parent.
+
+The subnet registry behaves as a subnet actor factory that offers users a convenient way of deploying instances of the reference implementation of the subnet actor in a network.
+
+The IPC actors are implemented using the diamond pattern to make it easier to upgrade them. The following sections describes how to upgrade each of these actors.
```

Comment:
It says "how to upgrade each of these actors." however there is only a section on Subnet Actor. What is the status of upgrading the gateway and registry?

Reply:
I will update the PR with an empty section for upgrading the gateway/registry which @snissn was going to fill out ([slack](https://filecoinproject.slack.com/archives/C06KWC57DRA/p1714494536816469?thread_ts=1714456806.300739&cid=C06KWC57DRA))